packages/twenty-front/src/modules/object-record/record-table/record-table-cell/hooks/__tests__/useMoveSoftFocusToCurrentCellOnHover.test.tsx

```diff
@@ -0,0 +1,141 @@
+import { act, renderHook } from '@testing-library/react';
+import { CallbackInterface, RecoilRoot } from 'recoil';
+
+import { RecordTableCellContext } from '@/object-record/record-table/contexts/RecordTableCellContext';
+import { RecordTableRowContext } from '@/object-record/record-table/contexts/RecordTableRowContext';
+import {
+  recordTableCell,
+  recordTableRow,
+} from '@/object-record/record-table/record-table-cell/hooks/__mocks__/cell';
+import { useMoveSoftFocusToCurrentCellOnHover } from '@/object-record/record-table/record-table-cell/hooks/useMoveSoftFocusToCurrentCellOnHover';
+import { RecordTableScope } from '@/object-record/record-table/scopes/RecordTableScope';
+import { TableCellPosition } from '@/object-record/record-table/types/TableCellPosition';
+import { TableHotkeyScope } from '@/object-record/record-table/types/TableHotkeyScope';
+
+const mockSoftFocusPositionState = {
+  key: 'softFocusPositionStateScopeMap__{"scopeId":"scopeId"}',
+};
+const mockSoftFocusActiveState = {
+  key: 'isSoftFocusActiveStateScopeMap__{"scopeId":"scopeId"}',
+};
+const mockIsSoftFocusOnTableCellFamilyState = {
+  key: 'isSoftFocusOnTableCellFamilyStateScopeMap__{"familyKey":{"column":1,"row":0},"scopeId":"scopeId"}',
+};
+const mockCurrentTableCellInEditModePositionState = {
+  key: 'currentTableCellInEditModePositionStateScopeMap__{"scopeId":"scopeId"}',
+};
+const mockIsTableCellInEditModeFamilyState = {
+  key: 'isTableCellInEditModeFamilyStateScopeMap__{"familyKey":{"column":1,"row":0},"scopeId":"scopeId"}',
+};
+const mockCurrentHotKeyScopeState = {
+  key: 'currentHotkeyScopeState',
+};
+
+const mockCallbackInterface = {
+  set: jest.fn(),
+  snapshot: {
+    getLoadable: (recoilValue: { key: string }) => ({
+      getValue: () => {
+        if (recoilValue.key === mockSoftFocusPositionState.key)
+          return { column: 1, row: 0 };
+        if (recoilValue.key === mockCurrentTableCellInEditModePositionState.key)
+          return { column: 1, row: 0 };
+        else if (recoilValue.key === mockIsTableCellInEditModeFamilyState.key)
+          return false;
+        else if (recoilValue.key === mockCurrentHotKeyScopeState.key)
+          return { scope: TableHotkeyScope.Table };
+      },
+    }),
+  },
+} as unknown as CallbackInterface;
+
+const setSoftFocusPosition = jest.fn();
+const setOnColumnsChange = jest.fn();
+const setHotkeyScope = jest.fn();
+
+jest.mock('recoil', () => ({
+  ...jest.requireActual('recoil'),
+  useRecoilCallback: (
+    callback: (
+      _interface: CallbackInterface,
+    ) => (newPosition: TableCellPosition) => void,
+  ) => callback(mockCallbackInterface),
+}));
```

Comment:
Testing functions that utilize Recoil's `useRecoilBack` for state updates can pose challenges. The asynchronous nature of Recoil makes it uncertain when the state has been updated, making it difficult to assert values accurately. To address this issue, I chose to mock the hook, providing greater control over the function's behavior and enabling more precise assertions on the new and old values.